app/src/main/java/com/odysee/app/MainActivity.java

```diff
@@ -86,14 +86,11 @@
 import com.google.android.exoplayer2.ui.PlayerView;
 import com.google.android.exoplayer2.upstream.cache.Cache;
 import com.google.android.gms.cast.framework.CastContext;
-import com.google.android.gms.tasks.OnCompleteListener;
-import com.google.android.gms.tasks.Task;
 import com.google.android.material.bottomnavigation.BottomNavigationView;
 import com.google.android.material.button.MaterialButton;
 import com.google.android.material.navigation.NavigationBarView;
 import com.google.android.material.snackbar.Snackbar;
 import com.google.android.material.textfield.TextInputEditText;
-import com.google.firebase.messaging.FirebaseMessaging;
```

Comment:
Everything on main will be available for all variants. That’s the common code folder. MainActivity.java must be moved into full AND foss, making changes on both of them depending on its product flavor.

/main -> everything common to play store and foss release, including AndroidManifest
/full -> any file which uses Play APIS, including AndroidManifest
/foss => any file which is not using Play APIs, including AndroidManifest

Reply:
I designed it that the refactoring will create helper classes in full and foss (the foss one is a stub), so the code that calls It can be kept the same.